reactive/kotlinx-coroutines-rx2/src/RxScheduler.kt

```diff
@@ -5,24 +5,87 @@
 package kotlinx.coroutines.rx2
 
 import io.reactivex.Scheduler
+import io.reactivex.disposables.Disposable
+import io.reactivex.disposables.Disposables
+import io.reactivex.plugins.RxJavaPlugins
 import kotlinx.coroutines.*
 import java.util.concurrent.TimeUnit
 import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
 
 /**
  * Converts an instance of [Scheduler] to an implementation of [CoroutineDispatcher]
  * and provides native support of [delay] and [withTimeout].
  */
 public fun Scheduler.asCoroutineDispatcher(): SchedulerCoroutineDispatcher = SchedulerCoroutineDispatcher(this)
 
+/**
+ * Converts an instance of [CoroutineDispatcher] to an implementation of [Scheduler].
+ */
+public fun CoroutineDispatcher.asScheduler(): Scheduler =
+        if (this is SchedulerCoroutineDispatcher) {
+            scheduler
+        } else {
+            DispatcherScheduler(this)
+        }
+
+private class DispatcherScheduler(private val dispatcher: CoroutineDispatcher) : Scheduler() {
+
+    private val scope = CoroutineScope(SupervisorJob())
+
+    override fun scheduleDirect(run: java.lang.Runnable): Disposable {
+        val decoratedRun = RxJavaPlugins.onSchedule(run)
+        val worker = createWorker()
+        worker.schedule(decoratedRun)
+        return worker
+    }
+
+    override fun scheduleDirect(run: java.lang.Runnable, delay: Long, unit: TimeUnit): Disposable {
+        val decoratedRun = RxJavaPlugins.onSchedule(run)
+        val worker = createWorker()
+        scope.launch {
```

Comment:
You must run this task in the scope of the worker's job that was just created. The way it is written right now, destroying this worker while it waits for the initial delay does not cancel this waiting job. Please, add a stress test for this problem -- schedule & immediately cancel millions of jobs. This code, before it is fixed, will get OutOfMemory error.

Reply:
@elizarov while it's not created with the job from ```launch```, the worker does create its own job upon instantiation. so it will get cancelled. Although it seems like a waste to create two Jobs so I'll change. Hopefully that makes sense.